File: cpp/1328.break-a-palindrome.cpp
/*
 * @lc app=leetcode id=1328 lang=cpp
 *
 * [1328] Break a Palindrome
 */
#include <string>
using namespace std;
// @lc code=start
class Solution
{
public:
    string breakPalindrome(string palindrome)
    {
        int len = palindrome.size();
        if (len == 1)
        {
            return "";
        }
        int pos = palindrome.find_first_not_of('a');
        if (pos == -1 || (len % 2) && pos == len / 2)
        {
            palindrome.pop_back();
            palindrome.push_back('b');
        }
        else
        {
            palindrome[pos] = 'a';
        }
        return palindrome;
    }
};
// @lc code=end

File: java/1328.break-a-palindrome.java
/*
 * @lc app=leetcode id=1328 lang=java
 *
 * [1328] Break a Palindrome
 */

// @lc code=start
class Solution {
    public String breakPalindrome(String palindrome) {
        int len = palindrome.length();
        if (len == 1) {
            return "";
        }
        int pos = 0;
        while (palindrome.charAt(pos) == 'a') {
            ++pos;
            if (pos >= len / 2) {
                pos = -1;
                break;
            }
        }
        if (pos == -1) {
            return palindrome.substring(0, len - 1) + 'b';
        } else {
            return palindrome.substring(0, pos) + 'a' + palindrome.substring(pos + 1);
        }
    }
}
// @lc code=end
